cmd/cli/version/version.go

```diff
@@ -104,13 +119,84 @@ func (oV *VersionOptions) Run(ctx context.Context, cmd *cobra.Command) error {
 	if !oV.ClientOnly {
 		serverVersion, err := util.GetAPIClient(ctx).Version(ctx)
 		if err != nil {
-			log.Ctx(ctx).Error().Err(err).Msgf("could not get server version")
-			return err
+			return fmt.Errorf("error running version: %w", err)
 		}
 
 		versions.ServerVersion = serverVersion
 		columns = append(columns, serverVersionColumn)
+
+		clientID, err := config.GetClientID()
+		if err != nil {
+			return fmt.Errorf("error getting UserID: %w", err)
+		}
+
+		updateCheck, err := checkForUpdates(ctx, versions.ClientVersion, versions.ServerVersion, clientID)
+		if err != nil {
+			log.Ctx(ctx).Warn().Err(err).Msg("Failed to perform update check")
+		} else {
+			versions.UpdateMessage = updateCheck.Message
+			versions.LatestVersion = updateCheck.Version
+			columns = append(columns, latestVersionColumn)
+
+			// Print the update message only if --output flag is not used
+			if oV.OutputOpts.Format == output.TableFormat {
+				fmt.Println(updateCheck.Message)
+			} else {
+				columns = append(columns, updateMessageColumn)
+			}
+		}
 	}
 
 	return output.OutputOne(cmd, columns, oV.OutputOpts, versions)
 }
+
+type serverResponse struct {
+	Version *models.BuildVersionInfo `json:"version"`
+	Message string                   `json:"message"`
+}
+
+func checkForUpdates(
+	ctx context.Context,
+	currentClientVersion, currentServerVersion *models.BuildVersionInfo,
+	clientID string,
+) (*serverResponse, error) {
+	u, err := url.Parse("http://update.bacalhau.org/version")
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to parse URL")
```

Comment:
```suggestion
		return nil, errors.Wrap(err, "failed to parse URL for update host")
```

a little bit more context gives us more of a clue of where to look when we see the error.  With more generic error messages they might appear more than once and leave us looking in the wrong place.